server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -109,8 +109,23 @@ public TransportFieldCapabilitiesAction(
 
     @Override
     protected void doExecute(Task task, FieldCapabilitiesRequest request, final ActionListener<FieldCapabilitiesResponse> listener) {
+        ActionListener<FieldCapabilitiesResponse> loggingListener = listener.delegateResponse((l, e) -> {
+            int statusCode = ExceptionsHelper.status(e).getStatus();
+            if (statusCode >= 500) {
+                logger.debug(
+                    "{} Exception in TransportFieldCapabilitiesAction for task: [{}] and request: [{}]. Exception: {}. Stack trace: {}",
+                    statusCode,
+                    task,
+                    request,
+                    e.getMessage(),
+                    ExceptionsHelper.stackTrace(e)
+                );
+            }
+            l.onFailure(e);
+        });
+
         // workaround for https://github.com/elastic/elasticsearch/issues/97916 - TODO remove this when we can
-        searchCoordinationExecutor.execute(ActionRunnable.wrap(listener, l -> doExecuteForked(task, request, l)));
+        searchCoordinationExecutor.execute(ActionRunnable.wrap(listener, l -> doExecuteForked(task, request, loggingListener)));
```

Comment:
If `doExecuteForked` itself throws something then it won't be logged with the code as written here. Also it's generally better not to capture listeners in lambdas and instead use the supplied argument (although it doesn't matter much here since other things are captured too):

```suggestion
        searchCoordinationExecutor.execute(ActionRunnable.wrap(loggingListener, l -> doExecuteForked(task, request, l)));
```